server/controllers/duels.js

```diff
@@ -594,10 +588,17 @@ async function updateDuelState(duel, username) {
     looser,
   });
 
-  return formatDuel(
+  const format = formatDuel(
     res.find((e) => e instanceof Array),
     username
   );
+
+  if (format.inProgress === 0) {
+    const currentDate = formatDate();
+    const sql = "UPDATE duel SET du_finished = ? WHERE du_id = ?;";
+    await queryPromise(sql, [currentDate, duel.id]);
```

Comment:
As this query doesn't have an effect on `format`, can we skip the `await` here?
```suggestion
    queryPromise(sql, [currentDate, duel.id]);
```

Reply:
No because at the end of this function we expect the duel to be updated, so if the following code requires this query to be executed, it might be a problem to remove it !